app/services/recurring_contribution/subscriptions/processor.rb

```diff
@@ -31,6 +34,20 @@ def new_credit_card_id
     register_error(:credit_card_key, :credit_card_invalid)
   end
 
+  def charge(subscription)
+    RecurringContribution::Subscriptions::Create.process(subscription)
+  end
+
+  def charge_scheduled_for_today?
+    (subscription.charging_day == DateTime.current.day) && with_save
+  end
+
+  def subscription_status
+    return :pending_payment if charge_scheduled_for_today?
```

Comment:
O retorno deste método depende mesmo do `with_save`, ou apenas do `charging_day`? Pq o `charge_scheduled_for_today?` verifica os dois.

Se sempre que a cobrança for para hoje depender do `with_save`, deixa assim mesmo. Se não, talvez fosse melhor ou renomear o método, ou extrair o `with_save`.